### PR TITLE
Convert PromptComponent class-based component to functional component…

### LINKED ISSUE #8779: [UI] Convert `PromptComponent` class-based component to functional component
### Current Behavior
PromptComponent is a class-based component.

### Desired Behavior
PromptComponents should be a functional component.

### Implementation
Convert all the Class-based components to new updated functional components. It is important here to convert a class-based component to a functional component you may need to convert some functions as well. Check if the function is used in other components or not.

Also, make sure to use the hooks while converting them. For example, you need to covert withNotify with useNotification, useRouter hook. Please check if a hook for functional components is available and use that.

Here is a quick blog that will help you get started with : [Link](https://www.digitalocean.com/community/tutorials/five-ways-to-convert-react-class-components-to-functional-components-with-react-hooks)

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Ishwar-Zatke**
Description:
Converted the class-based component of PromptCOmponent to functional based component.

This PR fixes 
#8779

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Ishwar-Zatke** [comment]
@leecalcote as there was conflict with #8794, i opened this new pr, will you look into this?

File: ui/components/PromptComponent.js [hunk 1]
```diff
@@ -60,130 +57,71 @@ const styles = (theme) => ({
...
 
-  handleCheckboxChange = () => {
-    this.setState((prevState) => ({
-      isChecked : !prevState.isChecked,
-    }));
-  };
-
-  getCheckboxState = () => {
-    return this.state.isChecked;
-  };
```
Comment (Yashsharma1911) [inline ui/components/PromptComponent.js:115]: Removing these functions will give errors in other modals that use checkbox, I wonder if you have any particular reason to remove them?

File: ui/components/PromptComponent.js [hunk 2]
```diff
@@ -60,130 +57,71 @@ const styles = (theme) => ({
...
-                      color="primary"
-                    />
-                  }
-                  label={
-                    <span className={classes.checkboxLabelStyle}>
-                      Do not show again
-                    </span>
-                  }
-                />
-              )}
```
Comment (Yashsharma1911) [inline ui/components/PromptComponent.js:161]: Yes this checkbox

**Comment by Yashsharma1911** [comment]
@Ishwar-Zatke merge conflict

**Comment by Yashsharma1911** [comment]
@Ishwar-Zatke Checking in, do you need any help?

**Pull request closed without merging**